### PR TITLE
Auto refresh AWS credentials

### PR DESCRIPTION
**Pull request by ayumi**
- Extract encrypt() and decrypt() from sync.js into cryptoUtil.js
- Extract refreshAWSCredentials() from sync.js into requestUtil.js
- requestUtil automatically detects expired credentials and requests new ones from the server

### CODE REVIEW

File: client/requestUtil.js [hunk 1]
```diff
@@ -1,27 +1,103 @@
 'use strict'
 
+const MAX_RETRIES = 1
+const NONCE_COUNTER = 0
```
Comment (diracdeltas) [inline client/requestUtil.js]: this should be imported from client/config.js so it doesn't need to be edited in more than one place when it changes

File: client/sync.js [hunk 2]
```diff
@@ -196,7 +136,14 @@ Promise.all([serializer.init(''), initializer.init(window.chrome)]).then((values
   logSync(`initialized userId ${clientUserId}`)
 })
   .then(() => {
-    return getAWSCredentials()
+    requester = new RequestUtil({
+      apiVersion: config.apiVersion,
+      credentialsBytes: null, // TODO: Start with previous session's credentials
+      keys: clientKeys,
+      serializer: clientSerializer,
+      syncServerUrl: config.serverUrl
```
Comment (diracdeltas) [inline client/sync.js]: this throws a 'missing serverUrl' error

File: client/sync.js [hunk 3]
```diff
@@ -47,68 +47,8 @@ const logSync = (message, logLevel = DEBUG) => {
...
-    .then((buffer) => {
-      requester = new RequestUtil(clientSerializer, new Uint8Array(buffer),
-        config.apiVersion, clientUserId)
-      if (!requester.s3) {
-        throw new Error('could not initialize AWS SDK')
-      }
-    })
-}
+const decrypt = cryptoUtil.Decrypt(clientSerializer, clientKeys.secretboxKey)
+const encrypt = cryptoUtil.Encrypt(clientSerializer, clientKeys.secretboxKey, conf.nonceCounter)
```
Comment (diracdeltas) [inline client/sync.js]: these functions need to be initialized after clientSerializer is initialized, otherwise clientSerializer is null.

**Comment by ayumi** [comment]
I made some changes based on your feedbacks. @diracdeltas could you take another look?

**Comment by diracdeltas** [comment]
thanks, lgtm!

**Pull request merged**